components/base-node.js

```diff
@@ -65,7 +67,43 @@ module.exports = {
        ar.push(item);
      }
      return ar;
-   } 
+   },
+
+   updatePromise: function(def) {
```

Comment:
I think we should avoid using the term update or updater at this layer to avoid confusion with the wrappers. Instead maybe we should call this something like vniComponent to indicate that it will be responsible for managing the vni states.